query-node/schemas/tag.graphql

```diff
@@ -0,0 +1,39 @@
+type tag @entity {
+  "Runtime entity identifier (EntityId)"
+  id: ID!
+  "The name of tag"
+  name: String!
+  "The description of tag"
+  description: String!
+  "The type of tag"
+  type: String!
+  "Visibility of tag"
+  visibility: Boolean!
+}
+
+type tagToWorker @entity {
+  "Runtime entity identifier (EntityId)"
+  id: ID!
+  "Id of tag"
+  tagId: ID!
+  "Id of worker"
+  workerId: ID!
+}
+
+type tagToThread @entity {
+  "Runtime entity identifier (EntityId)"
+  id: ID!
+  "Id of tag"
+  tagId: ID!
+  "Id of worker"
+  threadId: ID!
+}
+
+type tagToProposal @entity {
+  "Runtime entity identifier (EntityId)"
+  id: ID!
+  "Id of tag"
+  tagId: ID!
+  "Id of proposal"
+  proposalId: ID!
+}
```

Comment:
Do we need those intermediate entities? I think having `Thread` or `Proposal` entities have `tags` relation directly makes more sense, WDYT?

Reply:
well... In the requirement, the tags verus thread/proposal are Many : one relationship. I checked this kind of implementation in the original code. For example, in proposals.graphql and proposalsEvents.graphql, the proposal contains the proposalStatusUpdates[] relation (one : many) by using @derivedFrom(field: proposal). The proposalStatusUpdatedEvent contains the proposal field to indicate which proposal is associated. 
If I follow this mechanism, each tag data needs to contain proposalid or threadid, and there will be happened same tag content with different proposalid or threadid. so I suggested to use intermediate mapping structure for it. also this will not apply any changes to proposal and thread associated code and can be integrated.  I hope to hear from you about this.

Reply:
Please refer to implementation of legacy forum tags:
1. https://github.com/Joystream/joystream/blob/8e88b27c27204d06f9930c9f1cac129ab4671147/query-node/schemas/forum.graphql#L118-L127
2. https://github.com/Joystream/joystream/blob/8e88b27c27204d06f9930c9f1cac129ab4671147/query-node/schemas/forum.graphql#L115
Let's follow the same approach